From 15ed39f532bf16bc3d51b7d23a48976848a301c5 Mon Sep 17 00:00:00 2001
From: Timur Tabi <timur@freescale.com>
Date: Fri, 7 May 2010 15:53:58 +0800
Subject: [PATCH 01/11] p4080ds: add local bus interrupt parent property to device trees

The localbus nodes in all of the P4080 DS hardware device trees are missing an
interrupt-parent node.  In some cases, the Hypervisor will display this warning:

[0] dt_lookup_irqs_nolock: localbus@ffe124000 has interrupts but no interrupt-parent

The solution is to add an interrupt-parent property to the localbus node.

Signed-off-by: Timur Tabi <timur@freescale.com>
[Original patch
"kernel-2.6.30-p4080ds-add-local-bus-interrupt-parent-prope.patch"
taken from FSL p4080 SDK 2.1 ISO image.
Remove the parts for p4080ds_36b.dts]
Integrated-by: Kevin Hao <kexin.hao@windriver.com>
---
 arch/powerpc/boot/dts/p4080ds.dts      |    1 +
 arch/powerpc/boot/dts/p4080ds_srio.dts |    1 +
 2 files changed, 2 insertions(+), 0 deletions(-)

diff --git a/arch/powerpc/boot/dts/p4080ds.dts b/arch/powerpc/boot/dts/p4080ds.dts
index 9985a17..5203312 100644
--- a/arch/powerpc/boot/dts/p4080ds.dts
+++ b/arch/powerpc/boot/dts/p4080ds.dts
@@ -1913,6 +1913,7 @@
 		compatible = "fsl,p4080-elbc", "fsl,elbc", "simple-bus";
 		reg = <0 0xfe124000 0 0x1000>;
 		interrupts = <25 2>;
+		interrupt-parent = <&mpic>;
 		#address-cells = <2>;
 		#size-cells = <1>;
 
diff --git a/arch/powerpc/boot/dts/p4080ds_srio.dts b/arch/powerpc/boot/dts/p4080ds_srio.dts
index beca2da..c4d6eb3 100644
--- a/arch/powerpc/boot/dts/p4080ds_srio.dts
+++ b/arch/powerpc/boot/dts/p4080ds_srio.dts
@@ -1872,6 +1872,7 @@
 		compatible = "fsl,p4080-elbc", "fsl,elbc", "simple-bus";
 		reg = <0 0xfe124000 0 0x1000>;
 		interrupts = <25 2>;
+		interrupt-parent = <&mpic>;
 		#address-cells = <2>;
 		#size-cells = <1>;
 
-- 
1.6.5.2

